File: CSharp/Yow.CoD.Finance.Domain/Contracts/Duration.cs
namespace Yow.CoD.Finance.Domain.Contracts
{
    public sealed class Duration
    {
        public Duration(int length, DurationUnit unit)
        {
            Length = length;
            Unit = unit;
        }

        public int Length { get; }
        public DurationUnit Unit { get; }
    }
}

File: java/src/main/java/com/leecampbell/cod/domain/contracts/Duration.java
package com.leecampbell.cod.domain.contracts;

public final class Duration {
    private final int length;
    private final DurationUnit unit;

    public Duration(int length, DurationUnit unit) {
        this.length = length;
        this.unit = unit;
    }

    public int length() {
        return this.length;
    }

    public DurationUnit Unit() {
        return this.unit;
    }

    public String toString() { 
        return "Duration{length:" + this.length + ", unit:" + this.unit + "}";
    } 
}